jetty-core/jetty-util/src/test/java/org/eclipse/jetty/util/URIUtilTest.java

```diff
@@ -221,6 +215,84 @@ public void testBadDecodePath(String encodedPath)
         assertThrows(IllegalArgumentException.class, () -> URIUtil.decodePath(encodedPath));
     }
 
+    // @checkstyle-disable-check : AvoidEscapedUnicodeCharactersCheck
+    public static Stream<Arguments> safeDecodePathSource()
+    {
+        return Stream.of(
+            Arguments.of("/foo/bar", "/foo/bar"),
+
+            // Simple encoding
+            Arguments.of("/f%20%6f/b%20r", "/f o/b r"),
+
+            // Raw UTF8 and unicode handling
+            Arguments.of("/foo/b\u00e4\u00e4", "/foo/bää"),
+            Arguments.of("/f\u20ac\u20ac/bar", "/f€€/bar"),
+
+            // Encoded UTF-8 unicode (euro)
+            Arguments.of("/f%e2%82%ac%E2%82%AC/bar", "/f€€/bar"),
+
+            // Encoded delimiters
+            Arguments.of("/foo%2fbar", "/foo%2Fbar"),
+            Arguments.of("/foo%252fbar", "/foo%252fbar"),
+            Arguments.of("/foo%3bbar", "/foo;bar"),
+            Arguments.of("/foo%3fbar", "/foo%3Fbar"),
```

Comment:
Why were the `f` in `%2f` capitalized?

Reply:
That's because of our canonical path mechanism. We produce an encoded path that can be string compared. I.e. we take out all variations, leave only characters that need encoding encoded and uniform capitalization